Reference/Searching/Examine/Quick-Start/index.md

```diff
@@ -66,208 +120,210 @@ Umbraco ships with three indexes:
-                        </li>
-                    }
-                }
-            </ul>
-        }
-        else
-        {
-            <p>No results found for query @searchTerm</p>
-        }
-    }
-    return;
+    ids = index
+        .Searcher
+        .CreateQuery("content")
+        .NodeTypeAlias("person")
+        .And()
+        .Field("nodeName", query)
+        .Execute()
+        .Select(x => x.Id);
 }
 ```
-At this point we have chosen to use the External index and it's searcher. 
 
 :::tip
 We reference the External index by it's alias "ExternalIndex". Umbraco has a set of 'Constants' that refer to the indexes that can be more convenient to use `Constants.UmbracoIndexes`. So, in the example here we could have used `Constants.UmbracoIndexes.ExternalIndexName` instead of "ExternalIndex".
 :::
 
-The `searcher` has a CreateQuery method, where you can choose to search content, media or members eg:
+The `Searcher` has a CreateQuery method, where you can choose to search content, media or members eg:
 ```csharp
-searcher.CreateQuery("content")
+Searcher.CreateQuery("content")
 ```
 
 From here you can see how we can chain together the logic to perform the search. In the example we are searching all `content` using the `person` Document Type, where the `nodeName` is equal to the search term that was typed in the input bar.
 
 ```csharp
-searcher.CreateQuery("content").NodeTypeAlias("person").And().Field("nodeName", searchTerm)
+Searcher.CreateQuery("content").NodeTypeAlias("person").And().Field("nodeName", searchTerm)
 ```
-### Execute the Search
 
-Finally calling `.Execute()` at the end of the query logic will trigger the search and return a set of matching search results that we can loop through.
+Calling `.Execute()` at the end of the query logic triggers the search and returns a set of matching search results, which we can loop through to get the IDs of the resulting content items.
+## Getting the content
+We want to retrieve the actual content from the IDs. For that we need the `UmbracoHelper`, so let's go ahead and inject that into our `SearchService`.
+```csharp
+private readonly IExamineManager _examineManager;
+private readonly UmbracoHelper _umbracoHelper;
+
+public SearchService(IExamineManager examineManager, UmbracoHelper umbracoHelper)
+{
+    _examineManager = examineManager;
+    _umbracoHelper = umbracoHelper;
+}
+```
+
+The final method looks like this:
+```csharp
+public IEnumerable<IPublishedContent> SearchContentNames(string query)
+{
+    IEnumerable<string> ids = Array.Empty<string>();
+    if (!string.IsNullOrEmpty(query) && _examineManager.TryGetIndex("ExternalIndex", out IIndex? index))
+    {
+        ids = index.Searcher.CreateQuery("content").NodeTypeAlias("person").And().Field("nodeName", query).Execute()
+            .Select(x => x.Id);
+    }
+
+    foreach (var id in ids)
+    {
+        yield return _umbracoHelper.Content(id);
+    }
+}
+```
```

Comment:
Hm.. maybe it would be friendly if we put the entire, finalized service implementation here. Something along these lines:

```suggestion
We want to retrieve the actual content from the IDs. For that we need the `UmbracoHelper`, which must be injected into our service as well. The final implementation of `SearchService` then looks like this.

```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using Examine;
using Umbraco.Cms.Core.Models.PublishedContent;
using Umbraco.Cms.Web.Common;
using Umbraco.Extensions;

namespace MyStarterKitSite.Services;

public class SearchService : ISearchService
{
    private readonly IExamineManager _examineManager;
    private readonly UmbracoHelper _umbracoHelper;
    
    public SearchService(IExamineManager examineManager, UmbracoHelper umbracoHelper)
    {
        _examineManager = examineManager;
        _umbracoHelper = umbracoHelper;
    }

    public IEnumerable<IPublishedContent> SearchContentNames(string query)
    {
        IEnumerable<string> ids = Array.Empty<string>();
        if (!string.IsNullOrEmpty(query) && _examineManager.TryGetIndex("ExternalIndex", out IIndex? index))
        {
            ids = index
                .Searcher
                .CreateQuery("content")
                .NodeTypeAlias("person")
                .And()
                .Field("nodeName", query)
                .Execute()
                .Select(x => x.Id);
        }
        
        foreach (var id in ids)
        {
            yield return _umbracoHelper.Content(id);
        }
    }
}
```
```

Reply:
Grea idea 👍